sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -4055,19 +4055,28 @@ class AstBuilder extends DataTypeAstBuilder with SQLConfHelper with Logging {
   }
 
   /**
-   * Create a [[ShowTableExtended]] command.
+   * Create a [[ShowTablesExtended]] or [[ShowTablePartition]] command.
    */
   override def visitShowTableExtended(
       ctx: ShowTableExtendedContext): LogicalPlan = withOrigin(ctx) {
-    val partitionKeys = Option(ctx.partitionSpec).map { specCtx =>
-      UnresolvedPartitionSpec(visitNonOptionalPartitionSpec(specCtx), None)
+    def createUnresolvedTable(
```

Comment:
There is already a `createUnresolvedTable` function in this file, can we reuse it?

Reply:
It seems difficult to reuse because
- In existing functions `createUnresolvedTable `, `ctx: IdentifierReferenceContext,` represents `namespace. tablename`, it is a complete name.
https://github.com/apache/spark/blob/e499ba80be6828dcf0c34d3f06f1cd12321ce5b8/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala#L2828-L2833

- But in our scenario, `nsCtx: IdentifierReferenceContext` represents namespace, and `patternCtx: StringLitContext represents` tablename.
<img width="809" alt="image" src="https://github.com/apache/spark/assets/15246973/c957c28f-b9ba-4714-971c-ced3e069bd13">
If we wang to reuse it, we must use `nsCtx + patternCtx` to construct an `IdentifierReferenceContext`.